rest_api/tests/unit/mock_validator.py

```diff
@@ -0,0 +1,327 @@
+class _StateListHandler(_MockHandler):
+    def __init__(self):
+        super().__init__(
+            client.ClientStateListRequest,
+            client.ClientStateListResponse,
+            Message.CLIENT_STATE_LIST_RESPONSE)
+
+    def handle(self, content):
+        request = self._parse_request(content)
+        state = _MockState()
+
+        head_id, leaves = state.get_leaves(request.address, request.head_id)
+        if leaves == None:
+            return self._response_proto(status=self._response_proto.NO_ROOT)
+        if leaves == []:
+            return self._response_proto(
+                status=self._response_proto.NO_RESOURCE,
+                head_id=head_id)
+
+        return self._response_proto(
+            status=self._response_proto.OK,
+            head_id=head_id,
+            leaves=leaves)
+
+
+class _StateGetHandler(_MockHandler):
+    def __init__(self):
+        super().__init__(
+            client.ClientStateGetRequest,
+            client.ClientStateGetResponse,
+            Message.CLIENT_STATE_GET_RESPONSE)
+
+    def handle(self, content):
+        request = self._parse_request(content)
+        state = _MockState()
+
+        head_id, value = state.get_leaf(request.address, request.head_id)
+        if value == None:
+            return self._response_proto(status=self._response_proto.NO_ROOT)
+        if value == '':
+            return self._response_proto(
+                status=self._response_proto.NO_RESOURCE,
+                head_id=head_id)
+
+        return self._response_proto(
+            status=self._response_proto.OK,
+            head_id=head_id,
+            value=value)
+
+
+class _BlockListHandler(_MockHandler):
+    def __init__(self):
+        super().__init__(
+            client.ClientBlockListRequest,
+            client.ClientBlockListResponse,
+            Message.CLIENT_BLOCK_LIST_RESPONSE)
+
+    def handle(self, content):
+        request = self._parse_request(content)
+        store = _MockBlockStore()
+
+        blocks = store.get_blocks(request.head_id)
+        if not blocks:
+            return self._response_proto(status=self._response_proto.NO_ROOT)
+
+        return self._response_proto(
+            status=self._response_proto.OK,
+            head_id=blocks[0].header_signature,
+            blocks=blocks)
+
+
+class _BlockGetHandler(_MockHandler):
+    def __init__(self):
+        super().__init__(
+            client.ClientBlockGetRequest,
+            client.ClientBlockGetResponse,
+            Message.CLIENT_BLOCK_GET_RESPONSE)
+
+    def handle(self, content):
+        request = self._parse_request(content)
+        store = _MockBlockStore()
+
+        block = store.get_block(request.block_id)
+        if not block:
+            return self._response_proto(
+                status=self._response_proto.NO_RESOURCE)
+
+        return self._response_proto(
+            status=self._response_proto.OK,
+            block=block)
+
+
+## RUN ##
+if __name__ == '__main__':
```

Comment:
This should be moved to another script if it is useful, we generally haven't put __main__ sections in library or test code historically.